src/main/java/mate/academy/service/impl/BookServiceImpl.java

```diff
@@ -0,0 +1,42 @@
+package mate.academy.service.impl;
+
+import java.util.List;
+import mate.academy.dao.BookDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.Book;
+import mate.academy.service.BookService;
+
+@Service
+public class BookServiceImpl implements BookService {
+    @Inject
+    private BookDao bookDao;
+
+    @Override
+    public Book create(Book book) {
+        return bookDao.create(book);
+    }
+
+    @Override
+    public Book findById(Long id) {
+        return bookDao.findById(id).orElseThrow(() ->
+                new DataProcessingException("Couldn't get book by id " + id,
+                        new RuntimeException("Driver not found in DB.")));

```

Comment:
```suggestion
        return bookDao.findById(id).orElseThrow(() ->
                new RuntimeException("Driver not found in DB."));

```

Reply:
Not fixed. It makes no sense to wrap two new exceptions one into one. You only need to throw out one.